src/se/element.rs

```diff
@@ -0,0 +1,1455 @@
+                    };
+                }
+
+                any!(false_: false => "false");
+                any!(true_:  true  => "true");
+
+                any!(i8_:    -42i8                => "-42");
+                any!(i16_:   -4200i16             => "-4200");
+                any!(i32_:   -42000000i32         => "-42000000");
+                any!(i64_:   -42000000000000i64   => "-42000000000000");
+                any!(isize_: -42000000000000isize => "-42000000000000");
+
+                any!(u8_:    42u8                => "42");
+                any!(u16_:   4200u16             => "4200");
+                any!(u32_:   42000000u32         => "42000000");
+                any!(u64_:   42000000000000u64   => "42000000000000");
+                any!(usize_: 42000000000000usize => "42000000000000");
+
+                serde_if_integer128! {
+                    any!(i128_: -420000000000000000000000000000i128 => "-420000000000000000000000000000");
+                    any!(u128_:  420000000000000000000000000000u128 => "420000000000000000000000000000");
+                }
+
+                any!(f32_: 4.2f32 => "4.2");
+                any!(f64_: 4.2f64 => "4.2");
+
+                any!(char_non_escaped: 'h' => "h");
+                any!(char_lt:   '<' => "&lt;");
+                any!(char_gt:   '>' => "&gt;");
+                any!(char_amp:  '&' => "&amp;");
+                any!(char_apos: '\'' => "&apos;");
+                any!(char_quot: '"' => "&quot;");
+                //TODO: add a setting to escape leading/trailing spaces, in order to
+                // pretty-print does not change the content
+                any!(char_space: ' ' => " ");
+
+                any!(str_non_escaped: "non-escaped string" => "non-escaped string");
+                any!(str_escaped: "<\"escaped & string'>" => "&lt;&quot;escaped &amp; string&apos;&gt;");
+
+                err!(bytes:
+                    BTreeMap::from([("#any", Bytes(b"<\"escaped & bytes'>"))])
+                    => Unsupported("`serialize_bytes` not supported yet"));
+
+                any!(option_none: Option::<&str>::None);
+                any!(option_some: Some("non-escaped string") => "non-escaped string");
+                any!(option_some_empty_str: Some(""));
+
+                any!(unit: ());
+                any!(unit_struct: Unit);
+                any!(unit_struct_escaped: UnitEscaped);
+
+                any!(enum_unit: Enum::Unit => "<Unit/>");
+                err!(enum_unit_escaped:
+                    BTreeMap::from([("#any", Enum::UnitEscaped)])
+                    => Unsupported("character `<` is not allowed at the start of an XML name `<\"&'>`"));
+
+                any!(newtype: Newtype(42) => "42");
+                any!(enum_newtype: Enum::Newtype(42) => "<Newtype>42</Newtype>");
+
+                // Note that sequences of primitives serialized without delimiters!
+                any!(seq: vec![1, 2, 3] => "123");
```

Comment:
This seems rather dangerous and unintentional?  I'm missing something

Reply:
Unfortunately, that is. Yes, this can be unexpected, but this is consistent behaviour, and so it is preditable and can be explained. A possible solution here is to detect such a situation and return an error during serialization, I left the corresponding TODO for further improvements on line 55 in `content.rs`:
https://github.com/tafia/quick-xml/pull/490/files#diff-db7f64b5e8c9079f9ad5065b66f9c766bf2c32fed0d619ab2acc9e949248e166R55